packages/app/src/app/pages/common/Modals/SubscriptionCancellation/SubscriptionCancellation.tsx

```diff
@@ -0,0 +1,160 @@
+import track from '@codesandbox/common/lib/utils/analytics';
+import { Button, IconButton, Stack, Text } from '@codesandbox/components';
+import { TEAM_FREE_LIMITS } from 'app/constants';
+import { useCreateCustomerPortal } from 'app/hooks/useCreateCustomerPortal';
+import { useWorkspaceAuthorization } from 'app/hooks/useWorkspaceAuthorization';
+import { useWorkspaceSubscription } from 'app/hooks/useWorkspaceSubscription';
+import { useActions, useAppState } from 'app/overmind';
+import React from 'react';
+import styled from 'styled-components';
+
+const StyledListItem = styled('li')`
+  font-weight: 500;
+  font-size: 16px;
+  line-height: 1;
+  color: #ef7a7a;
+`;
+
+export const SubscriptionCancellationModal: React.FC = () => {
+  const { activeTeamInfo } = useAppState();
+  const { isTeamAdmin } = useWorkspaceAuthorization();
+  const { hasActiveTeamTrial } = useWorkspaceSubscription();
+  const { modalClosed } = useActions();
+  const [loadingCustomerPortal, createCustomerPortal] = useCreateCustomerPortal(
+    {
+      team_id: isTeamAdmin ? activeTeamInfo?.id : undefined,
+    }
+  );
+  // TO DO: if we need this info more often, extract
+  // to a custom useWorkspaceUsage similar to
+  // useWorkspaceLimits.
+  const teamUsage = activeTeamInfo?.usage;
+
+  const getEditorsLabel = () => {
+    if (teamUsage && teamUsage.editorsQuantity > TEAM_FREE_LIMITS.editors) {
+      const lostSeats = teamUsage.editorsQuantity - TEAM_FREE_LIMITS.editors;
+      return `-${lostSeats} editor seat${lostSeats === 1 ? '' : 's'}`;
+    }
+
+    return `Limited to ${TEAM_FREE_LIMITS.editors} editors seats`;
+  };
+
+  // Private sandboxes are a Pro feature only. If the
+  // team downgrades, these will become restricted.
+  // Public sandboxes do not get restricted on
+  // downgrading but are limited to a certain amount
+  // if the team does not have an active subscription.
+  const getSandboxesLabel = () => {
+    const restrictedSandboxesCount =
+      (teamUsage?.privateProjectsQuantity ?? 0) -
+      TEAM_FREE_LIMITS.private_repos;
+    if (restrictedSandboxesCount > 0) {
+      return `${restrictedSandboxesCount} sandbox${
+        restrictedSandboxesCount === 1 ? '' : 'es'
+      } will be restricted`;
+    }
+
+    return `Limited to ${TEAM_FREE_LIMITS.public_sandboxes} public sandboxes`;
+  };
+
+  // Private repositories are a Pro feature only. If the
+  // team downgrades, these will become restricted.
+  // Public repositories do not get restricted on
+  // downgrading but are limited to a certain amount
+  // if the team does not have an active subscription.
+  const getRepositoriesLabel = () => {
+    const restrictedRepositoriesCount =
+      (teamUsage?.privateProjectsQuantity ?? 0) -
+      TEAM_FREE_LIMITS.private_repos;
+    if (restrictedRepositoriesCount > 0) {
+      return `${restrictedRepositoriesCount} repositor${
+        restrictedRepositoriesCount === 1 ? 'y' : 'ies'
```

Comment:
should we use a pluralize utility for this? to make sure we don't spread too many of this tiny operations around the codebase and accidentally introduce spelling errors at some point?

Reply:
[`Intl.PluralRules` maybe](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/PluralRules)?

Reply:
Didn't want to add a library so created a util that takes the word, the count, and the suffixes: [packages/app/src/app/utils/pluralize.ts](https://github.com/codesandbox/codesandbox-client/pull/7195/files#diff-43ab7d886acc5095a8c0e69869cb0f82f241eabe44b409eb6f5d7e4042c9a004)